### PR TITLE
Added ontology objects for application dependencies

### PR DESCRIPTION
**Pull request by oxisto**
Currently they are hardcoded, we need to update owl2java at some point again. It will then displayed nicely in the UI as well. I added a section to the README with an example.

Screenshot 2023-11-01 at 10 25 38

### CODE REVIEW

File: voc/related.go [hunk 1]
```diff
@@ -41,9 +41,10 @@ func (v VirtualMachine) Related() []string {
 		list = append(list, string(b))
 	}
 
-	if v.BootLogging != nil {
-		list = append(list, v.BootLogging.Related()...)
-	}
+	// Boot/os logging in Azure points to a URL, not a Resource ID, this is broken!
```
Comment (lebogg) [inline voc/related.go:44]: I guess this change caused some tests to fail (see build error)

File: voc/library.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+// Library is an entity in our Cloud ontology. It encapsulates the (source) code of a library, similar to an application.
+// TODO(oxisto): Add this to the ontology and auto-generate it
+type Library struct {
+	*Resource
+	Functionalities     []*Functionality `json:"functionalities"`
+	ProgrammingLanguage string           `json:"programmingLanguage"`
+	TranslationUnits    []ResourceID     `json:"translationUnits"`
+	Dependencies        []ResourceID     `json:"dependencies"`
+}
```
Comment (lebogg) [inline voc/library.go:36]: I would prefer some small comments to clarify what they mean, e.g. for `Functionalities` and `TranslationUnits`. But, on the other hand, they would probably be overwritten by the auto generation anyway.

**Review by lebogg — COMMENTED** [review]
Good but fix tests

File: voc/related.go (hunk 1 above)
Comment (oxisto) [reply voc/related.go:44]: Ah good point. Should I remove it from the Azure discoverer then? Basically the discoverer is doing it incorrectly.

File: voc/library.go (hunk 2 above)
Comment (oxisto) [reply voc/library.go:36]: Yes. I think we had a beta branch of the auto-generator that generates also comments, but development there has stalled for a while.

File: voc/library.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package voc
+
+// Library is an entity in our Cloud ontology. It encapsulates the (source) code of a library, similar to an application.
+// TODO(oxisto): Add this to the ontology and auto-generate it
+type Library struct {
```
Comment (lebogg) [inline voc/library.go:30]: Do we want to have sth. like `LibraryType = [Library(, Application), Resource]` here as well?

File: voc/translation_unit_declaration.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// This file is part of Clouditor Community Edition.
+
+package voc
+
+var TranslationUnitDeclarationType = []string{"TranslationUnitDeclaration", "Resource"}
+
+// TranslationUnitDeclaration is an entity in our Cloud ontology. It refers to https://fraunhofer-aisec.github.io/cpg/CPG/specs/graph/#translationunitdeclaration
+type TranslationUnitDeclaration struct {
+	*Resource
+	Name string `json:"name"`
```
Comment (lebogg) [inline voc/translation_unit_declaration.go:35]: We have `Name` in the anonymous filed `Resource` already - with the same json tag. Will this be a problem?

File: api/discovery/experimental.proto [hunk 5]
```diff
@@ -45,7 +45,8 @@ service ExperimentalDiscovery {
   // the resource graph.
   //
   // Note: THIS API IS EXPERIMENTAL AND SUBJECT TO CHANGE
-  rpc UpdateResource(UpdateResourceRequest) returns (UpdateResourceResponse) {
+  rpc UpdateResource(UpdateResourceRequest)
+      returns (clouditor.discovery.v1.Resource) {
```
Comment (lebogg) [inline api/discovery/experimental.proto:49]: Yes :+1:

File: voc/related.go [hunk 6]
```diff
@@ -117,3 +117,19 @@ func (s Storage) Related() []string {
...
+	}
+
+	for _, tu := range a.TranslationUnits {
+		list = append(list, string(tu))
+	}
+
+	list = append(list, a.Resource.Related()...)
+
+	return list
+}
```
Comment (lebogg) [inline voc/related.go:135]: Isn't Compute a related field as well?

File: voc/translation_unit_declaration.go (hunk 4 above)
Comment (oxisto) [reply voc/translation_unit_declaration.go:35]: That is true, its redundant

**Pull request merged**